lib/fog/google/models/storage_json/file.rb

```diff
@@ -0,0 +1,154 @@
+        end
+
+        # TODO: Verify
+        def body=(new_body)
+          attributes[:body] = new_body
+        end
+
+        attr_reader :directory
+
+        # TODO: Verify
+        def copy(target_directory_key, target_file_key)
+          requires :directory, :key
+          service.copy_object(directory.key, key, target_directory_key, target_file_key)
+          target_directory = service.directories.new(:key => target_directory_key)
+          target_directory.files.get(target_file_key)
+        end
+
+        # TODO: Verify
+        def destroy
+          requires :directory, :key
+          begin
+            service.delete_object(directory.key, key)
+          rescue Excon::Errors::NotFound
+          end
+          true
+        end
+
+        # TODO: Verify
+        remove_method :metadata=
+        def metadata=(new_metadata)
+          metadata.merge!(new_metadata)
+        end
+
+        # TODO: Verify
+        remove_method :owner=
+        def owner=(new_owner)
+          if new_owner
+            attributes[:owner] = {
+              :entity => new_owner["entity"],
+              :entityId  => new_owner["entityId"]
+            }
+          end
+        end
+
+        # TODO: Verify
+        def public=(new_public)
+          if new_public
+            @predefined_acl = "publicRead"
+          else
+            @predefined_acl = "private"
+          end
+          new_public
+        end
+
+        # TODO: Verify
+        def public_url
+          requires :directory, :key
+
+          acl = service.get_object_acl(directory.key, key).body
+          access_granted = acl["items"].detect { |entry| entry["entity"] == "allUsers" && entry["role"] == "READER" }
+
+          if access_granted
+            if directory.key.to_s =~ /^(?:[a-z]|\d(?!\d{0,2}(?:\.\d{1,3}){3}$))(?:[a-z0-9]|\.(?![\.\-])|\-(?![\.])){1,61}[a-z0-9]$/
+              "https://#{directory.key}.storage.googleapis.com/#{key}"
+            else
+              "https://storage.googleapis.com/#{directory.key}/#{key}"
+            end
+          end
+        end
+
+        # TODO: Verify
+        def save(options = {})
+          requires :body, :directory, :key
+          if options != {}
+            Fog::Logger.deprecation("options param is deprecated, use acl= instead [light_black](#{caller.first})[/]")
+          end
+          options["contentType"] = content_type if content_type
+          options["predefinedAcl"] ||= @predefined_acl if @predefined_acl # predefinedAcl may need to be in parameters
+          options["acl"] ||= @acl if @acl # Not sure if you can provide both acl and predefinedAcl
+          options["cacheControl"] = cache_control if cache_control
+          options["contentDisposition"] = content_disposition if content_disposition
+          options["contentEncoding"] = content_encoding if content_encoding
+          options["md5Hash"] = content_md5 if content_md5
+          options["crc32c"] = crc32c if crc32c
```

Comment:
@Temikus I'm afraid i don't have enough knowledge to help regarding this.
@icco is the man you are looking for. =)